[lldb] Use the terminal height for paging editline completions

JDevlieghere opened the pull request:
Currently, we arbitrarily paginate editline completions to 40 elements. On large terminals, that leaves some real-estate unused. On small terminals, it's pretty annoying to not see the first completions. We can address both issues by using the terminal height for pagination.

This builds on the improvements of #116456.

JDevlieghere committed:
[lldb] Use the terminal height for paging editline completions

Currently, we arbitrarily paginate editline completions to 40 elements.
On large terminals, that leaves some real-estate unused. On small
terminals, it's pretty annoying to not see the first completions. We can
address both issues by using the terminal height for pagination.

JDevlieghere commented on lldb/source/Core/Debugger.cpp:
```diff
@@ -919,7 +937,11 @@ Debugger::Debugger(lldb::LogOutputCallback log_callback, void *baton)
       m_collection_sp->GetPropertyAtIndexAsOptionValueUInt64(
           ePropertyTerminalWidth);
   term_width->SetMinimumValue(10);
-  term_width->SetMaximumValue(1024);
```
I removed the maximum value here because I didn't want to duplicate it for the terminal height. If this bothers folks I can do that in a separate commit. I don't think it makes sense to arbitrarily limit the upper bound in the age of ultra-wide monitors.

JDevlieghere committed:
Update TestCompletion.py

labath commented on lldb/source/Host/common/Editline.cpp:
```diff
@@ -1000,14 +1010,17 @@ PrintCompletion(FILE *output_file,
       if (position + description_length < max_length) {
         fprintf(output_file, "%.*s\n", static_cast<int>(description_length),
                 line.data());
+        lines_printed++;
```
It seems like this could be a problem if we cross the page boundary in the middle of a multi-line description. The worst part is that given that the code above checks for strict equality, this situation might end up cause us to dump everything in a single go.

JDevlieghere replied:
Ah good point. I considered the situation where we would potentially exceed the page size if the last entry was a multiline one (and deemed that acceptable) but I didn't account for it in the check. We can make that a `>=`.

labath replied:
I suppose its not the end of the world, though it that case, it might be better to bail out here (and skip printing the rest of the description) instead of pushing the first completion off the screen (and if it fit on one line, the user wouldn't even know that it was there).

If this was c++20 I guess we could make this a coroutine which continues where it left off for the next page.

Up to you, I guess...

JDevlieghere committed:
Account for printed lines exceeding the max height

labath commented on lldb/source/Host/common/Editline.cpp:
```diff
@@ -1026,21 +1045,15 @@ void Editline::DisplayCompletions(
-
   size_t cur_pos = 0;
   while (cur_pos < results.size()) {
     size_t remaining = results.size() - cur_pos;
     size_t next_size = all ? remaining : std::min(page_size, remaining);
```
I think you don't need this now, and could always just pass the remainder of the list.

JDevlieghere committed:
Address Pavel's feedback

 - Always pass the whole list.
 - Truncate multi-line descriptions if they're about to overflow.

JDevlieghere committed:
Appease the formatter
